app/src/main/java/de/westnordost/streetcomplete/quests/surface/AddCyclewayPartSurface.kt

```diff
@@ -29,7 +37,17 @@ class AddCyclewayPartSurface(o: OverpassMapDataAndGeometryApi, r: ResurveyInterv
 
     override fun createForm() = AddPathSurfaceForm()
 
-    override fun applyAnswerTo(answer: String, changes: StringMapChangesBuilder) {
-        changes.updateWithCheckDate("cycleway:surface", answer)
+    override fun applyAnswerTo(answer: DetailSurfaceAnswer, changes: StringMapChangesBuilder) {
+        when (answer) {
+            is SurfaceAnswer -> {
+                changes.updateWithCheckDate("cycleway:surface", answer.value)
+                changes.deleteIfExists("source:cycleway:surface")
```

Comment:
Why is `source:cycleway:surface` only deleted when providing a detailed surface answer and not if a "generic" surface answer is provided? Same for the other classes that do this.
It could actually move out of the `when` no?

Reply:
Right, it should be changed